modules/redis/redis-worker.c

```diff
@@ -229,12 +300,18 @@ redis_worker_connect(LogThreadedDestWorker *s)
   RedisDriver *owner = (RedisDriver *) self->super.owner;
 
   if (self->c && check_connection_to_redis(self))
-    return TRUE;
-
-  if (self->c)
-    redisFree(self->c);
+    {
+      return TRUE;
+    }
+  else if(self->c)
+    {
+      redisReconnect(self->c);
```

Comment:
Why do we need this change ? Preferably move it into a different commit and explain in the commit message the reason behind this change.